SSE Extension: Default `sse-swap="message"` to keep setup simple

scriptogre opened the pull request:
## Summary

I wish this minimal setup just worked™:


```html
<div sse-connect="/events">Loading…</div>
```
```html
data: <span>I'm an unnamed event!</span>
```

**Expectation:** 
Payload from `data:` swaps directly into `<div sse-connect=...>` (using default swap strategy `innerHTML`).

**Reality:**
Unnamed events are ignored, unless you add `sse-swap="message"` to `sse-connect` element.


## Why?

I understand [browsers treat unnamed SSE events as `message` by default](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events#listening_for_message_events), but with this change we could:
1. Match the extension's default with the browser's default
2. Match the extension's behaviour with the behavior of `hx-get`/`hx-post` (where `hx-target="this"` is implicit)
3. Remove the gotcha & keep simple scenarios simple.


## Implementation

Add `sse-swap="message"` by default on the `sse-connect` element (unless there's any nested `sse-swap` attributes).


## Edge Cases

### Nested swaps

The pattern of using nested `sse-swap` attributes should continue to work (default is not applied in this scenario).

```html
<div sse-connect="/events">
  <div sse-swap="foo"></div>
  <div sse-swap="bar"></div>
</div>
```


### Triggering Server Callbacks

The pattern of triggering server callbacks should continue to work normally because the default `message` listener only catches unnamed events.

```html
<div hx-ext="sse" sse-connect="/event_stream">
    <div hx-get="/chatroom" hx-trigger="sse:chatter">
        ...
    </div>
</div>
```

---

Htmx version: 2.0.6
Used extension(s) version(s): 2.2.2

## Checklist

* [x] I have read the [contribution guidelines](https://github.com/bigskysoftware/htmx-extensions/blob/main/CONTRIBUTING.md)
* [x] I ran the test suite locally (`npm run test`) and verified that it succeeded

scriptogre committed:
feat(sse): Add default behavior for message event in SSE handling

AdamVig commented:
Would this work for multiple nested `sse-swap`s, like this?
```html
<div sse-connect="/events">
  <div sse-swap="foo"></div>
  <div sse-swap="bar"></div>
</div>
```
I'm wondering if this pull request would implicitly add `sse-swap="message"` to the parent `<div>`, which I think would break this pattern.

scriptogre commented:
That's a very good concern! Thanks for catching that for me. Let me test it out and see.

scriptogre committed:
fix(sse): Prevent default message behavior when nested sse-swap exists

The sse-swap attribute now defaults to "message" only when an element has
  sse-connect but no sse-swap AND no nested sse-swap elements. This preserves
  the common pattern of using parent elements as containers for multiple
  sse-swap children without conflicting message event listeners.

scriptogre commented:
The latest commit should handle it gracefully. I also added an explicit test for the scenario you described.

scriptogre commented:
This PR would also make #180 more neat.
